application/dos_api_gateway/dos_api_gateway.py

```diff
@@ -1,8 +1,12 @@
 from json import loads, dumps
 from typing import Dict, Any
 from aws_lambda_powertools.utilities.typing import LambdaContext
+from aws_lambda_powertools import Logger
 
+logger = Logger()
 
+
+@logger.inject_lambda_context(correlation_id_path="headers.x_correlation_id")
```

Comment:
How to handle if they don't pass x_correlation_id in headers. do we need to implement any validation?

Reply:
its optional, although we should probably check if the correlation id is provided and if not set one so we have the tracking at least ill do this now. if the header is not provided it continues gracefully

Reply:
in fact ill raise a subsequent PR for this